install markdown dependency for authoring in markdown

jldugger opened the pull request:
So restructured text is great, but discussion with Alex suggested being able to write posts in Markdown would be more conducive to getting students blogging. I agree, and we should allow people to write using it because it has the following benefits:

 - the language is simpler
 - they already use it for work on github
 - it's widely adopted by reddit, stackoverflow, etc.
 - the terse markup allows reviewers to focus on content rather than form
 - the failure case does not generally break the entire build

This PR adds the markdown processor, which is all I think is required to start allowing markdown documents -- pelican should autodetect them. Based on extension (.md) _I think_.

ramereth commented:
@jldugger have you verified locally that this is the only thing required to enable support? Have you actually created some test .md files to see if it works?

jldugger commented:
Yes. in branch jldugger/markdown-demo, I changed vim-tricks-ftw.rst to vim-tricks-ftw.md, changed the metadata header slightly, and the page still renders.
